backend/src/od.rs

```diff
@@ -0,0 +1,148 @@
+use geo::{BoundingRect, Contains, MultiPolygon, Point};
+use geojson::GeoJson;
+use nanorand::{Rng, WyRand};
+use serde::{Deserialize, Serialize};
+use utils::Mercator;
+
+use crate::{IntersectionID, MapModel};
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct ZoneID(pub usize);
+
+/// Origin/destination demand data, representing driving trips between two zones.
+#[derive(Serialize, Deserialize)]
+pub struct DemandModel {
+    pub zones: Vec<Zone>,
+    // (zone1, zone2, count), with count being the number of trips from zone1 to zone2
+    pub desire_lines: Vec<(ZoneID, ZoneID, usize)>,
+}
+
+impl DemandModel {
+    /// Turn all of the zones into Mercator. Don't do this when originally building and serializing
+    /// them, because that process might not use exactly the same Mercator object.
+    pub fn finish_loading(&mut self, mercator: &Mercator) {
+        for zone in &mut self.zones {
+            mercator.to_mercator_in_place(&mut zone.geometry);
+            let bbox = zone.geometry.bounding_rect().unwrap();
+            zone.x1 = (bbox.min().x * 100.0) as i64;
+            zone.y1 = (bbox.min().y * 100.0) as i64;
+            zone.x2 = (bbox.max().x * 100.0) as i64;
+            zone.y2 = (bbox.max().y * 100.0) as i64;
+        }
+    }
+
+    pub fn make_requests(&self, map: &MapModel) -> Vec<(IntersectionID, IntersectionID, usize)> {
+        info!(
+            "Making requests from {} zones and {} desire lines",
+            self.zones.len(),
+            self.desire_lines.len()
+        );
+
+        // TODO Plumb through UI
+        // To speed up the impact calculation, how many specific requests per (zone1, zone2)? If
+        // true, just do one, but weight it by count.
+        let fast_sample = true;
+
+        let mut rng = WyRand::new_seed(42);
+        let mut requests = Vec::new();
+
+        for (zone1, zone2, raw_count) in &self.desire_lines {
+            let (iterations, trip_count) = if fast_sample {
+                (1, *raw_count)
+            } else {
+                (*raw_count, 1)
+            };
+
+            for _ in 0..iterations {
+                let pt1 = self.zones[zone1.0].random_point(&mut rng);
+                let pt2 = self.zones[zone2.0].random_point(&mut rng);
+                if let (Some(i1), Some(i2)) = (
+                    map.closest_intersection
+                        .nearest_neighbor(&pt1)
+                        .map(|obj| obj.data),
+                    map.closest_intersection
+                        .nearest_neighbor(&pt2)
+                        .map(|obj| obj.data),
+                ) {
+                    if i1 != i2 {
+                        requests.push((i1, i2, trip_count));
+                    }
+                }
+            }
+        }
+        requests
+    }
+
+    pub fn to_gj(&self, map: &MapModel) -> GeoJson {
+        // Per (from, to) pair, how many trips?
```

Comment:
I'm not necessarily set on this as the best representation for the debug UI, just a starter